dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/dubboserver/DubboServer.java

```diff
@@ -0,0 +1,1277 @@
+        return this;
+    }
+
+    public boolean isInitialized() {
+        return initialized.get();
+    }
+
+    public boolean isStarted() {
+        return started.get();
+    }
+
+    public boolean isStartup() {
+        return startup.get();
+    }
+
+    public boolean isShutdown() {
+        return shutdown.get();
+    }
+
+    public DubboServer stop() throws IllegalStateException {
+        destroy();
+        return this;
+    }
+
+    /**
+     * Add an instance of {@link EventListener}
+     *
+     * @param listener {@link EventListener}
+     * @return {@link DubboServer}
+     */
+    public DubboServer addEventListener(EventListener<?> listener) {
+        eventDispatcher.addEventListener(listener);
+        return this;
+    }
+
+    /**
+     * export {@link MetadataService}
+     */
+    private void exportMetadataService() {
+        metadataServiceExporter.export();
+    }
+
+    private void unexportMetadataService() {
+        if (metadataServiceExporter != null && metadataServiceExporter.isExported()) {
+            metadataServiceExporter.unexport();
+        }
+    }
+
+    public DubboServer exportAsync() {
+        this.exportAsync = true;
+        return this;
+    }
+
+    public DubboServer referAsync() {
+        this.referAsync = true;
+        return this;
+    }
+
+
+    public DubboServer export(ServiceConfigBase serviceConfigBase) {
+        if (serviceConfigBase != null) {
+            exportService(serviceConfigBase);
+        }
+        return this;
+    }
+
+    public DubboServer exportAll() {
+        exportServices();
+        return this;
+    }
+
+    public DubboServer refer(ReferenceConfigBase referenceConfigBase) {
+        if (referenceConfigBase != null) {
+            referService(referenceConfigBase);
+        }
+        return this;
+    }
+
+    public DubboServer referAll() {
+        referServices();
+        return this;
+    }
+
+    public DubboServer unexport(ServiceConfigBase serviceConfigBase) {
+        if (serviceConfigBase != null) {
+            unexportService(serviceConfigBase);
+        }
+        return this;
+    }
+
+    public DubboServer unexportAll() {
+        unexportServices();
+        return this;
+    }
+
+    public DubboServer unrefer(ReferenceConfigBase referenceConfigBase) {
+        if (referenceConfigBase != null) {
+            unreferService(referenceConfigBase);
+        }
+
+        return this;
+    }
+
+    public DubboServer unreferAll() {
+        unreferServices();
+        return this;
+    }
+
+    public void init4ReferenceConfig(ReferenceConfig referenceConfig) {
+        if (referenceConfig.isInitialized()) {
+            return;
+        }
+
+        //use local variables to resonate with init() copied from ReferenceConfig
+        DubboBootstrap dubboBootstrap = referenceConfig.getDubboBootstrap();
+        if (dubboBootstrap == null) {
+            dubboBootstrap = DubboBootstrap.getInstance();
```

Comment:
The logic here is not clear, you should not initialize `DubboBootstrap` again, but use `this.dubboBootstrap`.